### PR TITLE
Implementation CRPS decoder

### LINKED ISSUE #2622: Implement CRPS decoder
### Describe the task. It can be a feature, documentation, etc.

Implement CRPS decoder

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

_No response_

### Area

- [ ] datasets, data readers, data preparation and transfer
- [ ] model
- [ ] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation
- [ ] performance

### PR DESCRIPTION
**Pull request by clessig**
## Description

Small cleanup of #2578 by @saibr

Usage:

```
decoder_ens_latent_perturbation : 
  num_members: 2    # 0 / 1 = disabled; >1 = number of ensemble members
  sigma_init: 0.2  # initial noise standard deviation
  sigma_learnable: True  # true = nn.Parameter; false = fixed buffer
```

## Issue Number

Closes #2622

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

**Comment by clessig** [comment]
@shmh40 : this is a slight cleanup of #2578 . Can you please have a look.

**Comment by shmh40** [comment]
```
decoder_ens_latent_perturbation : 
  num_members: 2    **# 0 / 1 = disabled;** >1 = number of ensemble members
  sigma_init: 0.2  # initial noise standard deviation
  sigma_learnable: True  # true = nn.Parameter; false = fixed buffer
  ```
  
  1  is not actually disabled right? I think this is what you want, but just checking.

File: src/weathergen/model/model.py [hunk 1]
```diff
@@ -745,94 +862,119 @@ def predict_decoders(
         Predict outputs at the specific target coordinates based on the input weather state and
         pre-training task and projects the latent space representation back to physical space.
 
+        If `latent_perturbation_num_members` > 1 (and `latent_perturbation_log_sigma` is set),
```
Comment (shmh40) [inline src/weathergen/model/model.py:865]: This latent_perturbation_num_members no longer exists. It has been replaced by decoder_ens_latent_perturbation.num_members

File: src/weathergen/model/model.py [hunk 2]
```diff
@@ -344,6 +344,23 @@ def __init__(self, cf: Config, sources_size, targets_num_channels, targets_coord
...
+        self.use_latent_perturbation = (
+            self.ens_latent_perturb is not None
+            and self.ens_latent_perturb.get("num_members", 0) >= 1
+        )
+
+        self.latent_perturbation_log_sigma = None
+        if self.use_latent_perturbation:
+            sigma_learnable = self.ens_latent_perturb.get("sigma_learnable", True)
+            self.latent_perturbation_log_sigma = nn.Parameter(
+                torch.zeros(1), requires_grad=sigma_learnable
```
Comment (shmh40) [inline src/weathergen/model/model.py:361]: So here we initialise this as 0, hence sigma is 1? And then it only takes a sensible value in reset_parameters?  Is reset_parameters always called? What happens when we continue or finetune a run? Why don't we just read sigma_init here?

File: src/weathergen/model/model.py [hunk 3]
```diff
@@ -344,6 +344,23 @@ def __init__(self, cf: Config, sources_size, targets_num_channels, targets_coord
...
+        # Latent-perturbation noise scale (learnable or fixed)
+        self.use_latent_perturbation = (
+            self.ens_latent_perturb is not None
+            and self.ens_latent_perturb.get("num_members", 0) >= 1
+        )
+
+        self.latent_perturbation_log_sigma = None
+        if self.use_latent_perturbation:
+            sigma_learnable = self.ens_latent_perturb.get("sigma_learnable", True)
+            self.latent_perturbation_log_sigma = nn.Parameter(
```
Comment (shmh40) [inline src/weathergen/model/model.py:360]: Since this is nn.Parameter, is this also subject to weight decay on all parameters? Do we want this?

**Review by shmh40 — COMMENTED** [review]
Please see comments.
